src/Components/Rating.js

```diff
@@ -0,0 +1,34 @@
+import React, { Component } from 'react'
+import PropTypes from "prop-types"
+import FullStar from "../star_full.svg"
+import HalfStar from "../star_half.svg"
+import EmptyStar from "../star_empty.svg"
+
+export default class Rating extends Component {
+  render() {
+    let stars = []

```

Comment:
dont need this to be a component class. can be component function only.

lets say this is a intensive computation, how can you use hooks to not compute this over and over on each rerender?

Reply:
I was thinking about something like this but i'm not sure..
```javascript
export default function Rating(props) {
  const [score] = useState(props.score)
  const [stars] = useState(function () {
    let stars = []
    let current = score
    for (let i = 0; i < 5; i++) {
      if (current >= 1) {
        stars.push(<img src={FullStar} alt="" key={i} />)
        current--
      } else if (current === 0.5) {
        stars.push(<img src={HalfStar} alt="" key={i} />)
        current--
      } else {
        stars.push(<img src={EmptyStar} alt="" key={i} />)
      }
    }
    return stars
  })
```